docker/Dockerfile

```diff
@@ -34,6 +36,12 @@ ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/openmpi/lib
 
 ENV NCCL_LAUNCH_MODE=GROUP
 
+#install nccl
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.0-1_all.deb && \
+    dpkg -i cuda-keyring_1.0-1_all.deb && \
+    apt update && \
+    apt install libnccl2 libnccl-dev
```

Comment:
why need to install nccl ? Do you meet any problem with current docker image?

Reply:
if not install nccl, execute "lmdeploy serve api_server" will be a not found error in latest image;

some person recurrence the bug in wechat group

Reply:
@ColorfulDick Hi, sorry. Can not reproduce.

Reply:
> @ColorfulDick Hi, sorry. Can not reproduce.

if not nccl,i even build the image successful,i show the pic in wechat group

Reply:
> @ColorfulDick Hi, sorry. Can not reproduce.

if no nccl,i can't build the image successful,i show the pic in wechat group;i suggest to add nccl and change cuda version to 12.2 in Dockerfile to support more people's envirments.
![微信图片_20240808165743](https://github.com/user-attachments/assets/e8c65f7e-765a-43e7-a938-0ef14bb6281d)

Reply:
hi, that's weird. Can you build with cu11 successfully?

Reply:
@RunningLeon I think we can refer to what vLLM does in https://github.com/vllm-project/vllm/blob/6dffa4b0a6120159ef2fe44d695a46817aff65bc/Dockerfile#L11
```
ARG CUDA_VERSION=12.4.1
#################### BASE BUILD IMAGE ####################
# prepare basic build environment
FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04 AS base
```

Reply:
> @ColorfulDick hi, that's weird. Can you build with cu11 successfully?

according to my build result, it's success to build the image in cuda 11.8 with not install nccl, but failed in cuda12.2 image  when execute  "sh ../generate.sh",it print "Could NOT find NCCL (missing: NCCL_INCLUDE_DIRS NCCL_LIBRARIES)" ; so i suggest to add nccl in Dockerfile to support more people' envirment at this stage;

Reply:
> @RunningLeon I think we can refer to what vLLM does in https://github.com/vllm-project/vllm/blob/6dffa4b0a6120159ef2fe44d695a46817aff65bc/Dockerfile#L11
> 
> ```
> ARG CUDA_VERSION=12.4.1
> #################### BASE BUILD IMAGE ####################
> # prepare basic build environment
> FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04 AS base
> ```

@lvhan028 could be a way. But it will make complicated and not controllable. 
1.Users can input any cuda version, which may not be covered in test.
2. pytorch needs to know cuda version and we have to add one more arg to set installing URL as in
https://github.com/InternLM/lmdeploy/blob/c685f777b397eb3d9fa3b2c0ac6dba2198586961/docker/Dockerfile#L24

Reply:
> > @ColorfulDick hi, that's weird. Can you build with cu11 successfully?
> 
> according to my build result, it's success to build the image in cuda 11.8 with not install nccl, but failed in cuda12.2 image when execute "sh ../generate.sh",it print "Could NOT find NCCL (missing: NCCL_INCLUDE_DIRS NCCL_LIBRARIES)" ; so i suggest to add nccl in Dockerfile to support more people' envirment at this stage;

ok. let me try with building cuda12.2.

Reply:
> > @ColorfulDick hi, that's weird. Can you build with cu11 successfully?
> 
> according to my build result, it's success to build the image in cuda 11.8 with not install nccl, but failed in cuda12.2 image when execute "sh ../generate.sh",it print "Could NOT find NCCL (missing: NCCL_INCLUDE_DIRS NCCL_LIBRARIES)" ; so i suggest to add nccl in Dockerfile to support more people' envirment at this stage;
@ColorfulDick hi, cannot reproduce. still building successfully with cuda12.2.

Reply:
@ColorfulDick I can accept installing nccl lib in dockerfile.